vertical-pod-autoscaler/pkg/admission-controller/resource/vpa/handler.go

```diff
@@ -133,9 +136,17 @@ func validateVPA(vpa *vpa_types.VerticalPodAutoscaler, isCreate bool) error {
 				}
 			}
 			for resource, min := range policy.MinAllowed {
+				if err := validate(resource, min); err != nil {
+					return err
+				}
 				max, found := policy.MaxAllowed[resource]
-				if found && max.Cmp(min) < 0 {
-					return fmt.Errorf("max resource for %v is lower than min", resource)
+				if found {
+					if err := validate(resource, max); err != nil {
```

Comment:
We should check if max has correct resolution in a loop:

```
for resource, min := range policy.MaxAllowed {
  if err := validate(resource, max); err != nil {
    //...
```

current implementation doesn't validate max when min is not specified.